cloud/amazon/ec2_vpc_route_table.py

```diff
@@ -322,16 +323,22 @@ def ensure_routes(vpc_conn, route_table, route_specs, propagating_vgw_ids,
 
     changed = routes_to_delete or route_specs_to_create
     if changed:
-        for route_spec in route_specs_to_create:
-            vpc_conn.create_route(route_table.id,
-                                  dry_run=check_mode,
-                                  **route_spec)
-
-        for route in routes_to_delete:
-            vpc_conn.delete_route(route_table.id,
-                                  route.destination_cidr_block,
-                                  dry_run=check_mode)
-    return {'changed': changed}
+        try:
+            for route_spec in route_specs_to_create:
+                    vpc_conn.create_route(route_table.id,
+                                          dry_run=check_mode,
+                                          **route_spec)
+
+            for route in routes_to_delete:
+                vpc_conn.delete_route(route_table.id,
+                                      route.destination_cidr_block,
+                                      dry_run=check_mode)
+
+        except EC2ResponseError as e:
+            if e.error_code == 'DryRunOperation':
+                pass
+
+    return {'changed': len(changed) > 0}
 
```

Comment:
There are two things addressed here:

1) The try/except block was implemented to allow the module to continue when the VPC API calls are made in check mode.  Revisiting the change its apparent that its incomplete though.  I probably should move the try/except block into each loop so subsequent dry run operations are attempted as well.  If this makese sense to you @wimnat I'll make the update.

2) The `changed` attribute in the returned dictionary was converted to a boolean.  http://docs.ansible.com/ansible/developing_modules.html seems to imply only booleans should be returned.  If non-boolean values are explicitly supported for python modules then I'll revert this change.